src/ai/.x/templates/openai-chat-w-asst-webpage-js/src/script.js

```diff
@@ -0,0 +1,512 @@
+}
+
+function chatPanelIsScrollAtBottom() {
+  let chatPanel = chatPanelGetElement();
+  let atBottom = Math.abs(chatPanel.scrollHeight - chatPanel.clientHeight - chatPanel.scrollTop) < 1;
+  return atBottom;
+}
+
+function chatPanelScrollToBottom() {
+  let chatPanel = chatPanelGetElement();
+  chatPanel.scrollTop = chatPanel.scrollHeight;
+}
+
+function chatPanelClear() {
+  let chatPanel = chatPanelGetElement();
+  chatPanel.innerHTML = '';
+}
+
+function logoGetElement() {
+  return document.getElementById("logo");
+}
+
+function logoShow() {
+  let logo = logoGetElement();
+  logo.style.display = "block";
+}
+
+function logoHide() {
+  let logo = logoGetElement();
+  logo.style.display = "none";
+}
+
+function markdownInit() {
+  marked.setOptions({
+    highlight: function (code, lang) {
+      let hl = lang === undefined || lang === ''
+        ? hljs.highlightAuto(code).value
+        : hljs.highlight(lang, code).value;
+      return `<div class="hljs">${hl}</div>`;
+    }
+  });
+}
+
+function markdownToHtml(markdownText) {
+  try {
+    return marked.parse(markdownText);
+  }
+  catch (error) {
+    return undefined;
+  }
+}
+
+function themeInit() {
+  let currentTheme = localStorage.getItem('theme');
+  if (currentTheme === 'dark') {
+    themeSetDark();
+  }
+  else if (currentTheme === 'light') {
+    themeSetLight();
+  }
+  toggleThemeButtonInit();
+}
+
+function themeIsLight() {
+  return document.body.classList.contains("light-theme");
+}
+
+function themeIsDark() {
+  return !themeIsLight();
+}
+
+function toggleTheme() {
+  if (themeIsLight()) {
+    themeSetDark();
+  } else {
+    themeSetLight();
+  }
+}
+
+function themeSetLight() {
+  if (!themeIsLight()) {
+    document.body.classList.add("light-theme");
+    localStorage.setItem('theme', 'light');
+
+    let iconElement = toggleThemeButtonGetElement().children[0];
+    iconElement.classList.remove("fa-toggle-on");
+    iconElement.classList.add("fa-toggle-off");
+  }
+}
+
+function themeSetDark() {
+  if (!themeIsDark()) {
+    document.body.classList.remove("light-theme");
+    localStorage.setItem('theme', 'dark');
+
+    let iconElement = toggleThemeButtonGetElement().children[0];
+    iconElement.classList.remove("fa-toggle-off");
+    iconElement.classList.add("fa-toggle-on");
+  }
+}
+
+function toggleThemeButtonGetElement() {
+  return document.getElementById("toggleThemeButton");
+}
+
+function toggleThemeButtonInit() {
+  let buttonElement = toggleThemeButtonGetElement();
+  buttonElement.addEventListener("click", toggleTheme);
+  buttonElement.addEventListener('keydown', toggleThemeButtonHandleKeyDown());
+}
+
+function toggleThemeButtonHandleKeyDown() {
+  return function (event) {
+    if (event.code === 'Enter' || event.code === 'Space') {
+      toggleTheme();
+    }
+  };
+}
+
+const titleUntitled = 'Untitled';
+
+function ThreadItem(id, created, metadata) {
+  this.id = id;
+  this.created = created;
+  this.metadata = metadata;
+}
```

Comment:
This function can be converted into a Class.

Reply:
i like this feedback, but i'll save this for a future PR where I do this for more samples in javascript across the board. Thanks!